src/Marten/Linq/SqlGeneration/JsonSelectClause.cs

```diff
@@ -103,15 +104,16 @@ public string Handle(DbDataReader reader, IMartenSession session)
 
             if (reader.Read())
             {
-                using var text = reader.GetTextReader(0);
-                builder.Write(text.ReadToEnd());
+                using var text = reader.GetStream(0);
+
+                builder.Write(text.GetStreamReader().ReadToEnd());
             }
 
             while (reader.Read())
             {
-                using var text = reader.GetTextReader(0);
+                using var text = reader.GetStream(0);
                 builder.Write(',');
-                builder.Write(text.ReadToEnd());
+                builder.Write(text.GetStreamReader().ReadToEnd());
```

Comment:
Let's reevaluate this one to see if there's a way to not have to read and create the string. See if there's a way to pipe the Stream bytes directly to the CommandBuilder